components/layout/flow/construct.rs

```diff
@@ -276,10 +290,27 @@ impl<'dom, 'style> BlockContainerBuilder<'dom, 'style> {
             Table::construct_anonymous(self.context, self.info, contents, propagated_data);
 
         if inline_table {
-            self.inline_formatting_context_builder.push_atomic(ifc);
+            self.ensure_inline_formatting_context_builder()
+                .push_atomic(ifc);
         } else {
             let table_block = ArcRefCell::new(BlockLevelBox::Independent(ifc));
-            self.end_ongoing_inline_formatting_context();
+
+            if let Some(inline_formatting_context) = self
+                .inline_formatting_context_builder
+                .take()
+                .and_then(|builder| {
+                    builder.finish(
+                        self.context,
+                        self.propagated_data,
+                        !self.have_already_seen_first_line_for_text_indent,
+                        self.info.is_single_line_text_input(),
+                        self.info.style.writing_mode.to_bidi_level(),
+                    )
+                })
```

Comment:
This same code happens in `finish()`. Maybe it's better to keep `end_ongoing_inline_formatting_context()`, update it with the new code, and call it from both places.